Formula/haskell-language-server.rb

```diff
@@ -40,23 +40,16 @@ def ghcs
         .sort_by(&:version)
   end
 
-  def newest_ghc
-    ghcs.max_by(&:version)
-  end
-
   def install
     system "cabal", "v2-update"
 
     ghcs.each do |ghc|
-      # for --enable-executable-dynamic flag, explained in
-      # https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html#support-for-template-haskell
-      args = ["-w", ghc.bin/"ghc", "--enable-executable-dynamic"]
-      system "cabal", "v2-install", *args, *std_cabal_v2_args
+      system "cabal", "v2-install", "--with-compiler=#{ghc.bin}/ghc", "--flags=-dynamic", *std_cabal_v2_args
```

Comment:
Would like feedback on this in case there is better way we could handle in Homebrew. As mentioned in previous comments, this will probably break Template Haskell support (though not sure of existing bottle support since upstream mentioned `--enable-executable-dynamic` was broken).

Reply:
So, to be clear, the `-dynamic` flag builds static binaries? That's confusing.

Reply:
`dynamic` is a `haskell-language-server` cabal flag so `+dynamic` enables while `-dynamic` disables it.

`--flags=+dynamic` just ends up running `ghc -dynamic` while building the binary while `--flags=-dynamic` uses default behavior (statically linked)